Features/BanCommand.cs

```diff
@@ -0,0 +1,89 @@
+using DSharpPlus.Commands;
+using DSharpPlus.Entities;
+using DSharpPlus.Commands.ContextChecks;
+using ArtcordAdminBot.Features.Helpers;
+using System.IO;
+using System.Net.Http;
+using System.Text.Json.Serialization;
+
+namespace ArtcordAdminBot.Features
+{
+    public class BanCommand
+    {
+        private readonly IDatabaseService _databaseService;
+        private readonly HttpClient _httpClient;
+
+        public BanCommand(IDatabaseService databaseService)
+        {
+            _databaseService = databaseService;
+            _httpClient = new HttpClient();
+        }
+
+        [Command("ban")]
+        [RequirePermissions(DiscordPermissions.BanMembers)]
+        public async Task BanAsync(CommandContext ctx, 
+        [System.ComponentModel.Description("The user to ban.")]DiscordUser targetUser, 
+        [System.ComponentModel.Description("The reason for the ban.")] string reason = "No reason provided", 
+        [System.ComponentModel.Description("The attachment of the reference image.")] DiscordAttachment? attachment = null, 
+        [System.ComponentModel.Description("The ID of the reference message.")] ulong? referenceMessageId = null, 
+        [System.ComponentModel.Description("Additional notes about the ban.")]string additionalNotes = "None", 
+        [System.ComponentModel.Description("Whether or not to delete the users messages from the last 7 days")] bool deleteMessages = false)
+        {
+            var time = TimeSpan.FromDays(7);
+            
+            // Attachment processing
+            string referenceImagePath = "None";
+            if (attachment != null)
+            {
+                // Download and save the image
+                string fileName = $"ban_reference_{targetUser.Id}_{DateTime.UtcNow.ToString("yyyy-MM-dd-HH-mm-ss")}.png";
```

Comment:
Include the Guild ID

```suggestion
                string fileName = $"ban_reference_{ctx.Guild!.Id}_{targetUser.Id}_{DateTime.UtcNow.ToString("yyyy-MM-dd-HH-mm-ss")}.png";
```

Also add the milliseconds at the end just to be sure it cannot come to collisions. Not sure how to put those into ToString format but look it up